cpp/include/cudf/hashing.hpp

```diff
@@ -34,42 +34,11 @@ namespace cudf {
  */
 using hash_value_type = uint32_t;
 
-/**
- * @brief Identifies the hash function to be used
- *
- */
-enum class hash_id {
-  HASH_IDENTITY = 0,   ///< Identity hash function that simply returns the key to be hashed
-  HASH_MURMUR3,        ///< Murmur3 hash function
-  HASH_SPARK_MURMUR3,  ///< Spark Murmur3 hash function
-  HASH_MD5             ///< MD5 hash function
```

Comment:
Since this is being removed from a public enum and the enum is moving to a different header, wondering if this PR should be marked as a breaking change.

Reply:
Agreed, this should be `breaking`. This enum should have been deprecated along with the `hash` function that uses it.

Reply:
If so, please defer merging this until there is Java work fixing such breaking is ready.

Reply:
spark-rapids-jni work tracked in https://github.com/NVIDIA/spark-rapids/issues/10487

Reply:
Spark dependency is now removed with https://github.com/NVIDIA/spark-rapids/pull/10652